src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
```

Comment:
PullProofAI

The `HttpExceptionFilter` class is missing error handling for cases where `err` is undefined. It's important to ensure that the code handles such scenarios gracefully to prevent runtime errors.

```diff
+     if (!err) {
+         err = 'Unknown error occurred';
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->